test/bench/bench.cpp

```diff
@@ -234,6 +236,17 @@ void load_benchmark(const fs::path& path, const std::string& name_prefix)
 
             case InputsReadingState::FuncName:
                 benchmark_case->func_name = std::move(l);
+                st = InputsReadingState::FuncSignature;
+                break;
+
+            case InputsReadingState::FuncSignature:
+                std::cout << l << std::endl;
+                if (l.find_first_of(":") != l.find_last_of(":"))
+                    throw std::runtime_error{"Multiple occurances of ':' found in signature"};
+                // Only allow i (i32) I (i64) v (void) as types
```

Comment:
Need to decide on a design here, the main question is about return types and no arguments cases:
a) Require colon and void type, e.g. `i:v`, `i:i`
b) Allow empty return type, no need for void, e.g. `i:`
c) Omit colon if no return type, e.g. `i`
d) Require void on input side if there no inputs, but there's an output, e.g. `v:i`
e) Allow empty arguments, e.g. `:i`

I'm not sure what is the best, but leaning towards c) and e).

cc @chfast @gumb0

Reply:
I like a) because everything is explicit and easier to parse

Reply:
In the case of a), what happens when there are no inputs? `:v` and `:i` ?

Reply:
1. If this is based on WAMR I would use what they have.
2. I would not use `v` at all to be consistent with wasm types, where it is `[]`.
3. I would keep `:` always for parsing. So `:` is also valid.

Reply:
It is not based on WAMR. Mentioned it here before: https://github.com/wasmx/fizzy/pull/318#discussion_r427904518

- wasm3 uses a version of this, but has a reverse notation where return value is first (e.g. `v(ii)`)
- emscripten uses a version of this: it has no separation of input vs output (e.g. `iiv`), but rather assumes the last type is either a type or void
- wamr uses a version of this, but unclear what they do for the return value

Reply:
@chfast are you saying these rules then:
- `:` no input, no output
- `i:` single i32 input, no output
- `:i` single i32 output, no input
- etc.

Reply:
Yes. But if this is something different than anything else, I don't care so much.

Reply:
We may want to improve/change/use it later on in the external API, then we can come back. All we need right now is just to make the benchmark tool more convenient to use and to support wamr.

Reply:
> In the case of a), what happens when there are no inputs? :v and :i ?

`v:v`, `v:i`
I guess it's d) or a) + d)